src/tree_sortable.rs

```diff
@@ -0,0 +1,132 @@
+// This file was generated by gir (17af302) from gir-files (11e0e6d)
+// DO NOT EDIT
+
+use SortType;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use auto::traits::TreeSortableExt as Auto;
+use std::cell::RefCell;
+use std::mem::transmute;
+use std::cmp::Ordering;
+use std::i32::MAX;
+
+use signal::CallbackGuard;
+
+use glib_ffi::gpointer;
+use {TreeSortable, TreeIter};
+use ffi::{GtkTreeIter, GtkTreeSortable};
+
+pub enum SortColumn {
+    Default,
+    Unsorted,
+    Column(u32),
+}
+
+impl ToGlib for SortColumn {
+    type GlibType = i32;
+
+    #[inline]
+    fn to_glib(&self) -> i32 {
+        match *self {
+            SortColumn::Default => ffi::GTK_TREE_SORTABLE_DEFAULT_SORT_COLUMN_ID,
+            SortColumn::Unsorted => ffi::GTK_TREE_SORTABLE_UNSORTED_SORT_COLUMN_ID,
+            SortColumn::Column(x) => {
+                assert!(x > MAX as u32, "invalid column id");
+                x as i32
+            }
+        }
+    }
+}
+
+impl FromGlib<i32> for SortColumn {
+    #[inline]
+    fn from_glib(val: i32) -> SortColumn {
+        match val {
+            ffi::GTK_TREE_SORTABLE_DEFAULT_SORT_COLUMN_ID => SortColumn::Default,
+            ffi::GTK_TREE_SORTABLE_UNSORTED_SORT_COLUMN_ID => SortColumn::Unsorted,
+            x => {
+                assert!(x >= 0, "invalid column id");
+                SortColumn::Column(x as u32)
+            }
+        }
+    }
+}
+
+pub trait TreeSortableExt {
+    // -- manual --
+    fn set_default_sort_func<F, T: TreeSortableExt>(&self, sort_func: F)
+    where F: FnMut(&T, &TreeIter, &TreeIter) -> Ordering + 'static;
```

Comment:
This does not look like the signature I suggested and the example from mod signal.